### PR TITLE
Support modern location expansions for run_binary

### PR DESCRIPTION
**Pull request by UebelAndre**
A common point of confusion I see around `run_binary` is that it's hard coded to only expand `$(location` values which in codebases I work in are otherwise completely eliminated due to it being described as "legacy"

> location: A synonym for either execpath or rootpath, depending on the attribute being expanded. This is legacy pre-Starlark behavior and not recommended unless you really know what it does for a particular rule. See [#2475](https://github.com/bazelbuild/bazel/issues/2475#issuecomment-339318016) for details.

If `execpath` is used instead as the appropriate alternative, the rule does no do any expansion and fails the action. This change adds support for expanding all available patterns whenever they're provided.

### CODE REVIEW

**Comment by UebelAndre** [comment]
@tetromino any chance you have time to review this?

File: rules/run_binary.bzl [hunk 1]
```diff
@@ -20,6 +20,22 @@ Runs a binary as a build action. This rule does not require Bash (unlike native.
 
 load("//lib:dicts.bzl", "dicts")
 
+_NOOPS = ("$(rlocationpath", "$(rootpath")
+_EXPANDS = ("$(execpath", "$(location")
+
+def _is_expandable(value):
```
Comment (tetromino) [inline rules/run_binary.bzl:26]: This check is misleading; if the template contains both `$(location ...)` and `$(rlocationpath ...)`, *both* will be expanded.

I think we ought to allow people to expand `$(rlocationpath` etc. - I don't see a use case for it, but going out of our way to filter it out complicates the code for no good reason.

**Review by tetromino — CHANGES_REQUESTED** [review]

File: rules/run_binary.bzl [hunk 2]
```diff
@@ -33,12 +49,12 @@ def _impl(ctx):
         # tokenization they would have to write args=["'a b'"] or args=["a\\ b"]. There's no
         # documented tokenization function anyway (as of 2019-05-21 ctx.tokenize exists but is
         # undocumented, see https://github.com/bazelbuild/bazel/issues/8389).
-        ctx.expand_location(a, tool_as_list) if "$(location" in a else a
+        ctx.expand_location(a, tool_as_list) if _is_expandable("$(location") else a
```
[the remark this answers is not in the document]
Comment (UebelAndre) [reply rules/run_binary.bzl:52]: Oops! Yes!

**Comment by UebelAndre** [comment]
@tetromino you available for another pass?

**Review by tetromino — APPROVED** [review]
LGTM. I've taken the liberty of updating the docs to mention both `$(execpath)` and `$(location)`.

**Pull request merged**